src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -352,49 +351,33 @@ void QMCCostFunctionBatched::checkConfigurations(EngineHandle& handle)
 
           handle.takeSample(energy_list, dlogpsi_array, dhpsioverpsi_array, ib);
         }
-
-        for (int ib = 0; ib < current_batch_size; ib++)
-        {
-          const int is = base_sample_index + ib;
-          auto etmp    = energy_list[ib];
-          opt_data.get_e0() += etmp;
-          opt_data.get_e2() += etmp * etmp;
-
-          RecordsOnNode[is][ENERGY_NEW]   = etmp;
-          RecordsOnNode[is][ENERGY_TOT]   = etmp;
-          RecordsOnNode[is][REWEIGHT]     = 1.0;
-          RecordsOnNode[is][ENERGY_FIXED] = h_list[ib].getLocalPotential();
-
-          const auto twf_dependent_components = h_list[ib].getTWFDependentComponents();
-          for (const OperatorBase& component: twf_dependent_components)
-            RecordsOnNode[is][ENERGY_FIXED] -= component.getValue();
-        }
       }
       else
+      { // Energy
+        energy_list = QMCHamiltonian::mw_evaluate(h_list, wf_list, p_list);
+      }
+      for (int ib = 0; ib < current_batch_size; ib++)
       {
-        // Energy
-        auto energy_list = QMCHamiltonian::mw_evaluate(h_list, wf_list, p_list);
-
-        for (int ib = 0; ib < current_batch_size; ib++)
-        {
-          const int is = base_sample_index + ib;
-          auto etmp    = energy_list[ib];
-          opt_data.get_e0() += etmp;
-          opt_data.get_e2() += etmp * etmp;
-
-          RecordsOnNode[is][ENERGY_NEW]   = etmp;
-          RecordsOnNode[is][ENERGY_TOT]   = etmp;
-          RecordsOnNode[is][ENERGY_FIXED] = h_list[ib].getLocalPotential();
```

Comment:
This also doesn't seem right. and can share code with above. So I made the change.